drivers/rc/lirc_dev.c

```diff
@@ -0,0 +1,1174 @@
+    {
+      fh = (FAR struct lirc_fh_s *)node;
+      if (LIRC_IS_TIMEOUT(sample) && !fh->send_timeout_reports)
+        {
+          continue;
+        }
+
+      if (lirc_buffer_write(fh->buffer, &sample, sizeof(unsigned)))
+        {
+          lirc_pollnotify(fh, POLLIN | POLLRDNORM);
+          nxsem_get_value(&fh->waitsem, &semcount);
+          if (semcount < 1)
+            {
+              nxsem_post(&fh->waitsem);
+            }
+        }
+    }
+
+  leave_critical_section(flags);
+}
+
+/****************************************************************************
+ * Name: lirc_scancode_event
+ *
+ * Description:
+ *   Lirc lowerhalf driver sends IR data to lirc upperhalf buffer, to
+ *   notify userspace to read IR data.
+ *
+ *   The type of data is struct lirc_scancode.
+ *
+ * Input Parameters:
+ *   lower  - A pointer to an instance of lower half lirc driver.
+ *   lsc    - The data of receiving from IR device
+ ****************************************************************************/
+
+void lirc_scancode_event(FAR struct lirc_lowerhalf_s *lower,
+                         FAR struct lirc_scancode *lsc)
+{
+  FAR struct lirc_upperhalf_s *upper = lower->priv;
+  FAR struct list_node *node;
+  FAR struct list_node *tmp;
+  FAR struct lirc_fh_s *fh;
+  irqstate_t flags;
+  int semcount;
+
+  lsc->timestamp = lirc_get_timestamp();
+
+  flags = enter_critical_section();
+  list_for_every_safe(&upper->fh, node, tmp)
+    {
+      fh = (FAR struct lirc_fh_s *)node;
+      if (lirc_buffer_write(fh->buffer, lsc, sizeof(*lsc)))
+        {
+          lirc_pollnotify(fh, POLLIN | POLLRDNORM);
+          nxsem_get_value(&fh->waitsem, &semcount);
+          if (semcount < 1)
+            {
+              nxsem_post(&fh->waitsem);
+            }
+        }
+    }
+
+  leave_critical_section(flags);
+}
+
+/****************************************************************************
+ * Name: lirc_sample_event
+ *
+ * Description:
+ *   Lirc lowerhalf driver sends raw IR data to lirc upperhalf buffer, to
+ *   notify userspace to read IR data.
+ *
+ *   The type of data is a sequence of pulse and space codes, as a seriers
+ *   of unsigned values.
+ *
+ *   The upper 8 bits determine the packet type, and the lower 24 bits the
+ *   payload.
+ *
+ * Input Parameters:
+ *   lower  - A pointer to an instance of lower half lirc driver.
+ *   sample - The data of receiving from IR device
+ ****************************************************************************/
+
+void lirc_sample_event(FAR struct lirc_lowerhalf_s *lower, unsigned sample)
+{
+  FAR struct lirc_upperhalf_s *upper = lower->priv;
+  FAR struct list_node *node;
+  FAR struct list_node *tmp;
+  FAR struct lirc_fh_s *fh;
+  irqstate_t flags;
+  int semcount;
+
+  flags = enter_critical_section();
+  list_for_every_safe(&upper->fh, node, tmp)
+    {
+      fh = (FAR struct lirc_fh_s *)node;
+      if (lirc_buffer_write(fh->buffer, &sample, sizeof(unsigned)))
```

Comment:
please use sizeof(unsigned int) as you did on read_mode2()

Reply:
done